synapse/static/client/register/js/register.js

```diff
@@ -1,67 +1,93 @@
 window.matrixRegistration = {
-    endpoint: location.origin + "/_matrix/client/api/v1/register"
+    endpoint: location.origin + "/_matrix/client/r0/register"
 };
 
 var setupCaptcha = function() {
-    if (!window.matrixRegistrationConfig) {
-        return;
-    }
-    $.get(matrixRegistration.endpoint, function(response) {
-        var serverExpectsCaptcha = false;
-        for (var i=0; i<response.flows.length; i++) {
-            var flow = response.flows[i];
-            if ("m.login.recaptcha" === flow.type) {
-                serverExpectsCaptcha = true;
-                break;
+    $.ajax({
+        url: matrixRegistration.endpoint,
+        method: "POST",
+        data: '{}',
```

Comment:
I think this is supposed to start with a basic request with no UIA, so just the desired username and password.

```suggestion
        data: JSON.stringify({username: user, password: pwd}),
```

... it's actually valid for a server to return a 200 to that, if it doesn't need recaptcha or whatever.